src/libraries/Common/src/System/Security/Cryptography/Asn1/Pkcs12/PfxAsn.manual.cs

```diff
@@ -2,12 +2,24 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System.Diagnostics;
+using System.Formats.Asn1;
+using System.Security.Cryptography.Asn1.Pkcs7;
 using System.Security.Cryptography.Pkcs;
+using System.Security.Cryptography.X509Certificates;
+using Internal.Cryptography;
+
+#if BUILDING_PKCS
+using Helpers = Internal.Cryptography.PkcsHelpers;
+#endif
 
 namespace System.Security.Cryptography.Asn1.Pkcs12
 {
     internal partial struct PfxAsn
     {
+        private const int MaxIterationWork = 300_000;
```

Comment:
I don't recall why we use 300k here but 600k on `X509Certificate.EnforceIterationCountLimit`. @krwq?

Reply:
I don't remember why this has separate limit but I suspect it has to do with the fact that we decrypt twice (null & empty) and after we're done OS will do the same work again so it may potentially quadruple the limit in the pessimistic case so it needs to be considerably smaller. Also I suspect this is not the limit people typically change. cc: @vcsjones